src/ssmDocument/ssm/ssmServer.ts

```diff
@@ -0,0 +1,537 @@
+            false
+        )
+
+        // Need all letters to be trigger characters for YAML completion
+        const triggerCharacters = [
+            '"',
+            '-',
+            '$',
+            'a',
+            'b',
+            'c',
+            'd',
+            'e',
+            'f',
+            'g',
+            'h',
+            'i',
+            'j',
+            'k',
+            'l',
+            'm',
+            'n',
+            'o',
+            'p',
+            'q',
+            'r',
+            's',
+            't',
+            'u',
+            'v',
+            'w',
+            'x',
+            'y',
+            'z',
+        ]
+        // tslint:enable: no-unsafe-any
+        const capabilities: ServerCapabilities = {
+            textDocumentSync: TextDocumentSyncKind.Incremental,
+            completionProvider: clientSnippetSupport
+                ? { resolveProvider: true, triggerCharacters: triggerCharacters }
+                : undefined,
+            hoverProvider: true,
+            documentSymbolProvider: true,
+            // tslint:disable-next-line: no-unsafe-any
+            documentRangeFormattingProvider: params.initializationOptions.provideFormatter === true,
+            colorProvider: {},
+            foldingRangeProvider: true,
+            selectionRangeProvider: true,
+        }
+        console.log('initialized.')
+        return { capabilities }
+    }
+)
+
+// The settings interface describes the server relevant settings part
+interface Settings {
+    aws?: {
+        ssmDocument?: {
+            ssm?: {
+                format?: { enable: boolean }
+                resultLimit?: number
+            }
+        }
+    }
+}
+
+namespace LimitExceededWarnings {
+    const pendingWarnings: { [uri: string]: { features: { [name: string]: string }; timeout?: NodeJS.Timeout } } = {}
+
+    export function cancel(uri: string) {
+        const warning = pendingWarnings[uri]
+        if (warning && warning.timeout) {
+            clearTimeout(warning.timeout)
+            // tslint:disable-next-line: no-dynamic-delete
+            delete pendingWarnings[uri]
+        }
+    }
+
+    export function onResultLimitExceeded(uri: string, maxResults: number, name: string) {
+        return () => {
+            let warning = pendingWarnings[uri]
+            if (warning) {
+                if (!warning.timeout) {
+                    // already shown
+                    return
+                }
+                warning.features[name] = name
+                warning.timeout.refresh()
+            } else {
+                warning = { features: { [name]: name } }
+                warning.timeout = setTimeout(() => {
+                    connection.sendNotification(
+                        ResultLimitReachedNotification.type,
+                        `${posix.basename(uri)}: For performance reasons, ${Object.keys(warning.features).join(
```

Comment:
I copied this part from the stepfunction [aslServer](https://github.com/aws/aws-toolkit-vscode/blob/feature/ssm-doc/src/stepFunctions/asl/aslServer.ts#L179) because I am doing something similar to what they are doing. `warning.features` will only have 1 key, which is the name.